### ISSUE TITLE
Wrong softmax results in current implementation

### ISSUE THREAD

**Issue by griegler** [rusty1s/pytorch_geometric#1354]
## 🐛 Bug

In the current [`softmax`](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py) implementation a bug was introduced that yields uniform probs because it uses just the max over the scatter. Previously, it was correctly implemented. Problem was introduced with `ptr` support [here](https://github.com/rusty1s/pytorch_geometric/commit/9615b9f904f8e4d4e65737587f28609bb888b1c4#diff-d2308e5ca4c12c61eac7ef5463d30be3). 

To resolve, just replace 

```
out = scatter(src, index, dim=0, dim_size=N, reduce='max')[index]
``` 

with 

```
out = out - scatter(src, index, dim=0, dim_size=N, reduce='max')[index]
``` 

Similar change is needed for the `ptr` code.

**Comment by rusty1s** [comment]
Oh ****, thanks for letting me know. Fixed in master.
